topostats/io.py

```diff
@@ -34,6 +34,54 @@
 # pylint: disable=too-many-lines
 
 
+# Sylvia: Ruff says too complex but I think breaking this out would be more complex.
+def dict_almost_equal(dict1: dict, dict2: dict, abs_tol: float = 1e-9):  # noqa: C901
```

Comment:
1 - Upon a quick glance, it `DeepDiff` seems like it might do what is needed

2 - When I wrote `dict_almost_equal`, I did add a test in `test_io.py`, line `172` in this branch.

Reply:
This function was already in the codebase, I just had to move it since I could not import it in the `tests/tracing/test_*.py` files since the function was in `tests/test_io.py` and pylint would not let me do `from ..test_io.py import dict_almost_equal` and python wouldn't recognise `from tests/test_io.py import dict_almost_equal` so I have moved it to `io.py` to allow importing in all tests.

Reply:
Self referential! Test code testing itself!